src/ui-test/tests/jbang.version.test.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { Workbench, VSBrowser, EditorView, WebDriver, after, before, ActivityBar, SideBarView, BottomBarPanel } from 'vscode-uitests-tooling';
+import * as path from 'path';
+import { CAMEL_ROUTE_YAML_WITH_SPACE, CAMEL_RUN_ACTION_LABEL, activateTerminalView, executeCommand, killTerminal, waitUntilTerminalHasText } from '../utils';
+import * as fs from 'node:fs';
+import { storageFolder } from '../uitest_runner';
+
+describe('JBang user preference version set test', function () {
+    this.timeout(120000);
+
+    let driver: WebDriver;
+    let editorView: EditorView;
+
+    let DEFAULT_JBANG: string;
+    const OLDER_JBANG_VERSION = '3.20.5';
+
+    const RESOURCES = path.resolve('src', 'ui-test', 'resources');
+
+    before(async function () {
+        driver = VSBrowser.instance.driver;
+        await VSBrowser.instance.openResources(path.join(RESOURCES));
+
+        DEFAULT_JBANG = await getJBangVersion();
+    });
+
+    after(async function () {
+        if (DEFAULT_JBANG.length > 0) {
+            await setJBangVersion(DEFAULT_JBANG);
+        } else {
+            resetUserSettings('camel.debugAdapter.JBangVersion');
+        }
+    });
+
+    describe('Different JBang versions', function () {
```

Comment:
could you please consider to move this test into existing `camel.settings.test.ts`? I think it could be done quite easily and it will avoid a code duplication or do you see some blockers to extend existing one? WDYT?